### PR TITLE
Upgrade ReactiveUI to the v8 alpha

### PR DESCRIPTION
**Pull request by jkoritzinsky**
The ReactiveUI v8 alpha version on NuGet has no dependencies on WPF in it's .NET 4.5 build, so we can finally use mainline ReactiveUI. I kept the Avalonia.ReactiveUI package because we still need to set our own scheduler. I decided to spin it off into its own package because nothing else in core depends on it and it would be nice to not have to depend on ReactiveUI for our core package.

### CODE REVIEW

File: build/Rx.props [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-    <PackageReference Include="System.Reactive.Linq" Version="3.0.0" />
-    <PackageReference Include="System.Reactive.PlatformServices" Version="3.0.0" />
-    <PackageReference Condition="'$(TargetFramework)' == 'net45'" Include="System.Reactive.Windows.Threading" Version="3.0.0" />
-    <PackageReference Condition="'$(TargetFramework)' == 'net461'" Include="System.Reactive.Windows.Threading" Version="3.0.0" />
+    <PackageReference Include="System.Reactive" Version="3.1.0" />
+    <PackageReference Include="System.Reactive.Core" Version="3.1.0" />
+    <PackageReference Include="System.Reactive.Interfaces" Version="3.1.0" />
+    <PackageReference Include="System.Reactive.Linq" Version="3.1.0" />
+    <PackageReference Include="System.Reactive.PlatformServices" Version="3.1.0" />
+    <PackageReference Condition="'$(TargetFramework)' == 'net45'" Include="System.Reactive.Windows.Threading" Version="3.1.0" />
```
Comment (kekekeks) [inline build/Rx.props]: `Condition="$(TargetFramework.EndsWith('net4'))"`

File: packages.cake [hunk 2]
```diff
@@ -111,6 +111,7 @@ public class Packages
         var SplatVersion = packageVersions["Splat"].FirstOrDefault().Item1;
         var SpracheVersion = packageVersions["Sprache"].FirstOrDefault().Item1;
         var SystemReactiveVersion = packageVersions["System.Reactive"].FirstOrDefault().Item1;
+        var ReactiveUIVersion = packageVersions["reactiveui"].FirstOrDefault().Item1;
```
Comment (kekekeks) [inline packages.cake]: I've started some kind of refactoring for dependency version detection, see
https://github.com/jkoritzinsky/Avalonia/blob/a0cfa2608b22c437f69563c1d09d0e333dcd4b8b/packages.cake#L250 and https://github.com/jkoritzinsky/Avalonia/blob/a0cfa2608b22c437f69563c1d09d0e333dcd4b8b/packages.cake#L37

We probably want to remove `Dep` overload from there and add some logic for own packages.

**Review by kekekeks — APPROVED** [review]
LGTM

File: build/Rx.props (hunk 1 above)
Comment (grokys) [inline build/Rx.props]: Shouldn't that be `StartsWith`?

**Review by grokys — APPROVED** [review]
Very excited to see this!

**Pull request merged**